src/util/findRouteByComponent.js

```diff
@@ -1,24 +1,23 @@
-export default function findRouteByComponent(component, routes) {
+export default function findRouteByComponent(component, routes, components) {
   const result = {};
-  for (const route of routes) {
-    if (route.component === component) {
-      result.route = route;
-      return result;
-    }
-    if (route.components) {
-      const foundNamedComponent = Object.keys(route.components)
+  components.some((matchedComponent, i) => {
+    if (typeof matchedComponent === 'object') {
+      Object.keys(matchedComponent)
         .some(key => {
-          const found = route.components[key] === component;
-          if (found) {
+          if (matchedComponent[key] === component) {
             result.name = key;
+            matchedComponent = matchedComponent[key]; // eslint-disable-line no-param-reassign
```

Comment:
Are we sure that this mutation does not cause any problems? Seems like we are modifying `renderProps` indirectly here, no risk in doing so?